_posts/2020-02-10-waku-current-state.md

```diff
@@ -0,0 +1,141 @@
+---
+layout: post
+name:  "Waku current state"
+title:  "Waku current state"
+date:   2020-02-10 12:00:00 +0800
+author: oskarth
+published: false
+permalink: /waku-current-state
+categories: research
+summary: A research log. Waku current state.
+image: /assets/img/TODOFIXME.png
+discuss: https://forum.vac.dev/t/TODOFIXME
+---
+
+TODO: Picture for bottleneck or Waku
+
+# Where we are at with Waku?
+
+Waku is our fork of Whisper where we address the shortcomings of Whisper in an iterative manner. We've seen in previous post that Whisper doesn't scale, and why. Here's an update on where we are at with Waku since that [last post](https://vac.dev/fixing-whisper-with-waku).
+
+## Current state
+
+**Specs:**
+
+We've cut the spec up into three components. These are:
+
+- Waku (main spec), currently in [version 0.2.0](https://specs.vac.dev/waku/waku.html)
+- Waku envelope data field, currently in [version 0.1.0](https://specs.vac.dev/waku/envelope-data-format.html)
+- Waku mailserver, currently in [version 0.2.0](https://specs.vac.dev/waku/mailserver.html)
+
+TODO: Update Waku spec to 0.3
+
+**Clients:**
+
+There are currently two clients that implement Waku, these are [Nimbus](https://github.com/status-im/nimbus/tree/master/waku) in Nim and [status-go](https://github.com/status-im/status-go) in Go.
+
+At the time of this writing the Nimbus client implements the spec fully, but lacks mail server and mail client capability. The status-go client implements everything except bridging mode, which is currently a work in progress.
```

Comment:
Not sure if this detail needs to be added, but the nim implementation also lacks rate limiting and confirmations. But these are optional so in theory it still complies to spec.

Reply:
added